File: CS/1021_Remove_Outermost_Parentheses.cs
public class Solution {
    public string RemoveOuterParentheses(string s) {
        StringBuilder result = new StringBuilder();
        int level = 0;
        foreach(char c in s){
            if(c == '('){
                level++;
                if(level > 1){
                    result.Append(c);
                }
            }
            else{
                level--;
                if(level > 0){
                    result.Append(c);
                }
            }
        }
        return result.ToString();
    }
}

File: Cpp/1021_Remove_Outermost_Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string result = "";
        int t = 0;
        for(char c : s){
            if(c == '('){
                t++;
                if(t != 1){
                    result += c;
                }
            }
            else{
                t--;
                if(t != 0){
                    result += c;
                }
            }
        }
        return result;
    }
};
